reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/TimedScheduler.java

```diff
@@ -87,13 +88,30 @@ Runnable wrapPeriodic(Runnable task) {
 	@Override
 	public Disposable schedule(Runnable task) {
 		this.submittedDirect.increment();
-		return delegate.schedule(wrap(task));
+		task = wrap(task);
+
+		try {
+			return delegate.schedule(task);
+		}
+		catch (RejectedExecutionException exception) {
+			((TimedRunnable) task).pendingSample.stop();
```

Comment:
Consider changing the signature of `wrap` and `wrapPeriodic` to return `TimedRunnable`, create a variable of that type instead of assigning to `task` and then you can avoid the cast here. I think it would be more elegant. This is justified in my opinion, as the `TimedScheduler` is a final class and the mentioned methods are just used here in the implementations and inside `TimedWorker`.